jac/jaclang/compiler/parser.py

```diff
@@ -3187,12 +3186,12 @@ def typed_filter_compare_list(self, kid: list[ast.AstNode]) -> ast.FilterCompr:
             else:
                 raise self.ice()
 
-        def filter_compare_item(self, kid: list[ast.AstNode]) -> ast.CompareExpr:
+        def filter_compare_item(self, _: None) -> ast.CompareExpr:
             """Grammar rule.
 
             filter_compare_item: name_ref cmp_op expression
             """
-            ret = self.compare(kid)
+            ret = self.compare(self.cur_nodes)
```

Comment:
Instead of this consume name_ref, cmp_op, and expression and build the node return it also remove the `self.ice()` bellow.